app/controllers/your-psp/post-flex-controller.js

```diff
@@ -1,25 +1,54 @@
 'use strict'
```

Comment:
Minor point, wondering if it is worth having a test for this file?
You have good coverage already - so not sure if it is worth it.

Reply:
I did consider this. There was no test for the controller before, just the Cypress tests for the whole thing. Obviously, it’s gotten more complex but the actual validation logic itself is well tested in unit tests (the validation rules) and the Cypress tests (the overall validation functionality). The main thing that isn’t really tested is the posting to connector and that hasn’t changed and would best be done as a Pact test. While I’m a little uncomfortable at the amount of testing for the 3DS Flex credentials setting overall, it is going to be used extremely rarely, so maybe it isn’t really worth going beyond what we already have.

Reply:
👍